test/e2e/providers/exec/docker.go

```diff
@@ -59,3 +59,35 @@ func RunDockerExecCommand(container string, cmd string, workdir string) (
 	// command is successful
 	return 0, string(stdoutBytes), string(stderrBytes), nil
 }
+
+func RunDockerPsCommand(flag string, argument string) (
```

Comment:
I'm not sure I understand the parameter list for this function. IMO base on your current usage, it would make more sense to have a single `filter string` parameter.

Reply:
I wanted to make this generic to accept any option/flag with docker ps command. When there is no argument, we pass empty string. For example, "docker ps -a". I can add this in function-level comment. 
https://docs.docker.com/engine/reference/commandline/ps/

Reply:
I don't think generic is the way to go here... And if you want to have a generic solution, the fact that you can only provide a single command-line flag is somewhat confusing. If we wanted to have a generic solution, we would just go with `RunDockerCommand(args ...string)`, but that wouldn't be very usable in my opinion... If you look at the Docker Go SDK (https://godoc.org/github.com/docker/docker/api/types#ContainerListOptions), they don't take a generic approach. Given that this function is only used in one place right now, and given that this is test code, I believe it makes sense to stick to a single `filter` parameter.

Reply:
Agreed. Missed the point that "docker ps" can take in multiple options.
Docker Go SDK options type makes sense. 
https://godoc.org/github.com/docker/docker/api/types#ContainerListOptions 
Sticking with single filter parameter. Updated the commit. Thanks for the comment.